bot/controllers/dice.js

```diff
@@ -0,0 +1,80 @@
+const BaseController = require('../baseController.js');
+const Command = require('../baseCommand.js');
+const util = require('apex-util');
+
+class DiceController extends BaseController {
+  constructor(message) {
+    // Call BaseController constructor
+    super(message);
+
+    // Aliasing 'this' as controller to allow for binding in actions
+    const controller = this;
+
+    // Array of all commands, see baseCommand.js for prototype
+    this.commands = [
+      new Command(
+        '!roll',
+        '!roll <roll_count><die_type>,[roll_count][die_type]',
+        'Rolls the dice',
+        'Rolls dice based on the user input. Must be in #D# format for multiple rolls or D# for a single roll. Example: 2D20 rolls two D20; D20 rolls one D20.',
+        this.rollAction.bind(controller),
+      ),
+    ];
+  }
+
+  // Perform the rolling of the dice
+  rollAction() {
+    const { message } = this;
+    const diceCommand = message.parsed[1].split(',');
```

Comment:
Magic Number (1) and Semantically diceCommand should be plural.

Reply:
Fixed these issues. For the magic number, this is present in all the other files as well, which is the only reason I assumed this instance of it was okay.